src/lib.rs

```diff
@@ -375,22 +379,25 @@ mod explicit {
                 PDI => {
                     if overflow_isolate_count > 0 {
                         overflow_isolate_count -= 1;
-                        continue
-                    }
-                    if valid_isolate_count == 0 {
-                        continue
-                    }
-                    overflow_embedding_count = 0;
-                    loop {
-                        // Pop everything up to and including the last Isolate status.
-                        match stack.vec.pop() {
-                            Some(Status { status: OverrideStatus::Isolate, .. }) => break,
-                            None => break,
-                            _ => continue
+                    } else if valid_isolate_count != 0 {
+                        overflow_embedding_count = 0;
+                        loop {
+                            // Pop everything up to and including the last Isolate status.
+                            match stack.vec.pop() {
+                                Some(Status { status: OverrideStatus::Isolate, .. }) => break,
+                                None => unreachable!(),
+                                _ => continue
+                            }
                         }
+                        valid_isolate_count -= 1;
+                    }
+                    let last = stack.last();
+                    result.levels[i] = last.level;
```

Comment:
It says to do this "In all cases", which I interpret as including cases that had `continue` before.